packages/client/hmi-client/src/services/concept.ts

```diff
@@ -83,9 +83,14 @@ async function getEntitySimilarity(
 	}
 }
 
-const getNameOfCurieCached = (cache: Map<string, string>, curie: string): string => {
+const getNameOfCurieCached = async (
+	curie: string,
+	cache: Map<string, string> = new Map<string, string>()
+): Promise<string> => {
```

Comment:
This is a rather confusing design pattern, you are calling this function without the cache functional parameter, you then instantiate the cache on the fly and it goes out of the scope when the function finishes.

So effective the changes you have made are not using cache at all. So either:
- we should create a cache-store at the service level, and we don't pass in a cache for the function invocation (which you don't seem to be doing anyway), or
- we don't use cache

Have a slight preference for option one. 

If you want to see a client side caching implementation,  we did it in the world modeler project, and there is an example here on how to hook things up:

https://github.com/uncharted-causemos/causemos/blob/56c57f0891d8d97051a34f411bedecf850922b0e/client/src/services/outputdata-service.ts#L38-L41